sockeye/inference.py

```diff
@@ -188,7 +201,15 @@ def sym_gen(source_seq_len: int):
 
             data_names = [C.SOURCE_NAME]
             label_names = []  # type: List[str]
-            return mx.sym.Group(decoder_init_states), data_names, label_names
+            # optionally predict length ratios
+
+            predicted_length_ratios = []
+            length_ratio_data_name = []
+            if hasattr(self, 'length_ratio'):
```

Comment:
why would it not have the attribute? Is this the correct way of checking for it?